nerfstudio/utils/tensor_dataclass.py

```diff
@@ -254,6 +323,14 @@ def _apply_fn_to_dict(self, dict_: Dict, fn: Callable, dataclass_fn: Optional[Ca
             if v is not None:
                 if isinstance(v, TensorDataclass) and dataclass_fn is not None:
                     new_dict[f] = dataclass_fn(v)
+                # This is the case when we have a custom dimensions tensor
+                elif (
+                    isinstance(v, torch.Tensor)
+                    and isinstance(self._field_custom_dimensions, dict)
+                    and f in self._field_custom_dimensions.keys()
```

Comment:
Am I wrong that `f in self._field_custom_dimensions.keys()` should be checked first? Since it takes priority to simply using `dataclass_fn(v)`.

Reply:
I think it should be fine since the custom dimensions only apply to torch.Tensor types and the case above this elif is for TensorDataclass objects, so this case is hit first if it's a torch.Tensor and is in our _field_custom_dimensions dict